portfolio/src/main/java/com/google/sps/servlets/DeletionServlet.java

```diff
@@ -62,6 +65,17 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     Key key = KeyFactory.createKey("Comment", body.getId());
 
     datastore.delete(key);
+
+    // Initialise a connection to Blobstore
+    BlobstoreService blobstore = BlobstoreServiceFactory.getBlobstoreService();
+
+    // Create a BlobKey from the provided BlobKey String
+    // In case of an invalid or non-existent BlobKey, the servlet will quit.
+    // This is expected, as some comments don't have images but should still
+    // be viable for deletion.
+    BlobKey blobKey = new BlobKey(body.getImageBlobstoreKey());
```

Comment:
What happens if there is an image but blobstore is down at the time you go to delete it?

Reply:
In that case, the image wouldn't be deleted. There's a hard dependency of the DeletionServlet on BlobStore, and the DataServlet is the same. Though I recognise that queueing all requests and then deleting them only when BlobStore is available would allow for me to avoid this dependency, I decided that, given the scale of the application, this redundancy wasn't necessary. The expected behaviour is for failed deletions to be silently ignored. At scale, this wouldn't be a viable option, but in the context of this application, it seemed like a reasonable compromise.

Reply:
If this was to be deployed as a real app, I'd recommend you reconsider, if a user uploaded an image then requested it to be deleted it could be a legal requirement to ensure that data is deleted.

A simpler solution would be to not silently fail, so the user knows there was a problem and they can retry the deletion.

Reply:
I've fixed this now 👍 Decided to go with the recommended option of prompting the user to retry. https://github.com/thomasogara/step-repository/pull/48/commits/fe1dc866da2ed392fd772b115c53f2a1a27c79a1